src/devices/Bmxx80/Bmxx80Base.cs

```diff
@@ -0,0 +1,215 @@
+            double var1 = ((adcTemperature / 16384.0) - (_calibrationData.DigT1 / 1024.0)) * _calibrationData.DigT2;
+            double var2 = ((adcTemperature / 131072.0) - (_calibrationData.DigT1 / 8192.0)) * _calibrationData.DigT3;
+            var2 *= var2 * _calibrationData.DigT3;
+
+            TemperatureFine = (int)(var1 + var2);
+
+            double temp = (var1 + var2) / 5120.0;
+            return Temperature.FromCelsius(temp);
+        }
+
+        /// <summary>
+        /// Reads an 8 bit value from a register.
+        /// </summary>
+        /// <param name="register">Register to read from.</param>
+        /// <returns>Value from register.</returns>
+        protected internal byte Read8BitsFromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                _i2cDevice.WriteByte(register);
+                byte value = _i2cDevice.ReadByte();
+                return value;
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        /// Reads a 16 bit value over I2C.
+        /// </summary>
+        /// <param name="register">Register to read from.</param>
+        /// <returns>Value from register.</returns>
+        protected internal ushort Read16BitsFromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                Span<byte> bytes = stackalloc byte[2];
+
+                _i2cDevice.WriteByte(register);
+                _i2cDevice.Read(bytes);
+
+                return BinaryPrimitives.ReadUInt16LittleEndian(bytes);
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        /// Reads a 24 bit value over I2C.
+        /// </summary>
+        /// <param name="register">Register to read from.</param>
+        /// <returns>Value from register.</returns>
+        protected internal uint Read24BitsFromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                Span<byte> bytes = stackalloc byte[4];
+
+                _i2cDevice.WriteByte(register);
+                _i2cDevice.Read(bytes.Slice(1));
+
+                return BinaryPrimitives.ReadUInt32LittleEndian(bytes);
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        protected Sampling ByteToSampling(byte value)
+        {
+            // Values >=5 equals UltraHighResolution.
+            if (value >= 5)
+            {
+                return Sampling.UltraHighResolution;
+            }
+
+            return (Sampling)value;
+        }
+
+        /// <summary>
+        /// Cleanup.
+        /// </summary>
+        public void Dispose()
```

Comment:
Can we instead implement the Dispose pattern here?

```c#
public void Dispose()
{
  Dispose(true);
  GC.SuppressFinalize(this);
}

protected virtual void Dispose(bool disposing)
{
  _i2cDevice?.Dispose();
  _i2cDevice = null;
}
```

That way if nested classes want to do their own dispose, all of the parent's ones will get called as well.

Reply:
Done! 😃